app/controllers/requests_controller.rb

```diff
@@ -40,16 +40,20 @@ def update
       if @request.save
         Transaction.create(request_id: @request.id, transaction_type: 'response')
         flash[:success] = "Thanks for being a good neighbor!"
-        redirect_to request_path(@request)
+        redirect_to request_path
       else
         flash[:error] = @request.errors.full_messages.join(', ')
         redirect_to "show"
       end
     else
       if current_user == @request.requester || current_user == @request.responder
-        @request.update_attribute(:is_fulfilled, true)
-        Transaction.create(request_id: @request.id, transaction_type: 'fulfillment')
-        redirect_to request_path(@request)
+        @request.is_fulfilled = true
+        if @request.save
+          Transaction.create(request_id: @request.id, transaction_type: 'fulfillment')
+          redirect_to request_path(@request)
+        else
+          redirect_to "show"
```

Comment:
I don't think you can do this. You have to `redirect_to` a named route or you have to `render` a view. Can't mix-and-match.